src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/plantation/PlantationModuleRegistry.java

```diff
@@ -0,0 +1,84 @@
+package com.minecolonies.coremod.colony.buildings.workerbuildings.plantation;
+
+import com.minecolonies.api.colony.buildings.workerbuildings.plantation.PlantationFieldType;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.plantation.modules.specific.*;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.EnumMap;
+import java.util.Map;
+
+/**
+ * Registry class for all different plantation module types.
+ */
+public class PlantationModuleRegistry
+{
+    /**
+     * Map containing the field types as well as the modules belonging to it.
+     */
+    private static final EnumMap<PlantationFieldType, PlantationModule> plantationModules = new EnumMap<>(PlantationFieldType.class);
```

Comment:
So, not a real registry. Cant this be part of our initial registry. Im a bit confused for the need for so many types and registries. Isnt it enough to have one Entry

BerryEntry, ThisEntry, ThatEntry, they also represent the types, and they also can get you the modules?